package.json

```
@@ -20,12 +22,20 @@
     "url": "https://github.com/lostfictions/znv/issues"
   },
   "files": [
-    "!dist/**/*.test.*",
-    "dist/"
+    "dist/",
+    "dist_module/"
   ],
+  "exports": {
+    ".": {
+      "import": "./dist_module/index.js",
+      "require": "./dist/index.js",
+      "types": "./dist_module/index.d.ts"
+    },
```

`types` is supposed to be declared first in this list ([source](https://nodejs.org/api/packages.html#community-conditions-definitions), [source](https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing))

```suggestion
    ".": {
      "types": "./dist_module/index.d.ts",
      "import": "./dist_module/index.js",
      "require": "./dist/index.js"
    },
```

further, are you sure the declarations emitted for esm are compatible with cjs like this? or do we need to do use the pattern of separating the esm and commonjs export maps?

```json
    "exports": {
        ".": {
            "import": {
                "types": "...",
                "default": "..."
            },
            "require": {
                "types": "...",
                "default": "..."
            },
        }
    },
```

---

Yes indeed, this should be done. I verified the change here https://arethetypeswrong.github.io/?p=bowlingx-znv%400.4.2 with a test version and it looks good :).